### ISSUE TITLE
Add Gradle support

### ISSUE THREAD

**Issue by ghost** [spring-projects/spring-petclinic#236]
Is there any chance that this project would be converted to the Gradle build system? Seems like Gradle is the future and it would really come in handy to see how such a project would utilize it.

I tried converting it but as this is the first real spring project I've dealt with, I cant wrap my head around all the configuration files :)

**Comment by Abrasha** [comment]
I think this is a good idea. Need some feedback from repository maintainers and I can take care of adding Gradle support.

**Comment by Dovchiproeng** [comment]
@Abrasha i also interested with it, if u need help I can give u a hand since i want to learn gradle too!

**Comment by arey** [comment]
I’m not sure Gradle is the future for build systems. You may refer the Baeldung Java in 2017 survey: http://www.baeldung.com/java-in-2017
Maven is slowly getting even more traction whereas Gradle remains stagnant.
Moreover, Java developers are more familiar with Maven.
In my opinion, Maven is more appropriate for sample applications like Petclinic. Application build should not be a difficulty. 

4 years ago, @whimet has developed a fork with Gradle: https://github.com/whimet/spring-petclinic It is not up to date. Thus I suggest to create a new Gradle fork that could take part of the Spring Petclinic community: https://github.com/spring-petclinic

**Comment by arey** [comment]
The Kotlin version of Spring Petclinic is using Gradle (with Kotlin instead of Grrovy): 
https://github.com/spring-petclinic/spring-petclinic-kotlin/blob/master/build.gradle.kts

**Comment by dsyer** [comment]
I would be happy to look at a pull request for a `build.gradle`. It's a bit of a pain to maintain 2 build systems in parallel, but it might be worth it since I don't think the `pom.xml` changes much.

**Comment by Lewiscowles1986** [comment]
Regardless of kotlin or Java most pom.xml files can be ported to Java using tools such as https://gist.github.com/jashatton/3237323

**Comment by bgalek** [comment]
I've made a PR with gradle support, if anybody still interested :)

**Comment by ShreyAgarwal11** [comment]
I also think gradle is the future as for maven it creates all the jars and files for all projects in one folder and if that folder gets deleted or goes corrupt then all the projects made on maven will get affected but that is not the case for gradle. In gradle individual folders are made for individual projects and therefore if one goes corrupt it wouldn't affect others. As for the survey is considered  maven is old and mostly everyone has used it at least once and therefore people are comfortable with it and would not like switching to gradle or maybe don't know the advantages of gradle but from what all I know about these two, I can say that gradle is better.

**Comment by Lewiscowles1986** [comment]
https://github.com/spring-projects/spring-petclinic/pull/385 now seems to be the PR to track. Not much movement since April, but given the review comments I can't blame them

- preserve maven
- add gradle

I think is the ask. But hidden in there is also "use gradle the way we want it..."

**Comment by Bwvolleyball** [comment]
Adding gradle to this project is very straight forward **_except_** for the wro4j configuration, which is proving to be quite complicated.  I've opened an issue in the [wro4j-gradle-plugin](https://github.com/wro4j/wro4j-gradle-plugin) hoping for some advice, but I'm not optimistic that this plugin will even work as expected with the current version of Gradle without some work.  

Would anyone happen to have a recommendation for a gradle less -> css plugin that can handle the inheritance from bootstrap? I noticed that the spring-petclinic-kotlin punted on this problem by directly including `petclinic.css`to the repository with the conversion to gradle...

**Comment by roy-gh-hm** [comment]
I ran `gradle init` in the root directory and got the following build.gradle; seems to work fine, though I can't find a `run` task...

```
/*
 * This file was generated by the Gradle 'init' task.
 */

plugins {
    id 'java'
    id 'maven-publish'
}

repositories {
    mavenLocal()
    maven {
        url = uri('https://repo.spring.io/snapshot')
    }

    maven {
        url = uri('https://repo.spring.io/milestone')
    }

    maven {
        url = uri('https://repo.maven.apache.org/maven2/')
    }
}

dependencies {
    implementation 'org.springframework.boot:spring-boot-starter-actuator:2.4.5'
    implementation 'org.springframework.boot:spring-boot-starter-cache:2.4.5'
    implementation 'org.springframework.boot:spring-boot-starter-data-jpa:2.4.5'
    implementation 'org.springframework.boot:spring-boot-starter-web:2.4.5'
    implementation 'org.springframework.boot:spring-boot-starter-validation:2.4.5'
    implementation 'org.springframework.boot:spring-boot-starter-thymeleaf:2.4.5'
    implementation 'javax.cache:cache-api:1.1.1'
    implementation 'org.ehcache:ehcache:3.9.2'
    implementation 'org.webjars:webjars-locator-core:0.46'
    implementation 'org.webjars:jquery:2.2.4'
    implementation 'org.webjars:jquery-ui:1.11.4'
    implementation 'org.webjars:bootstrap:3.3.6'
    implementation 'org.springframework.boot:spring-boot-devtools:2.4.5'
    runtimeOnly 'com.h2database:h2:1.4.200'
    runtimeOnly 'mysql:mysql-connector-java:8.0.23'
    testImplementation 'org.springframework.boot:spring-boot-starter-test:2.4.5'
}

group = 'org.springframework.samples'
version = '2.4.5'
description = 'petclinic'
java.sourceCompatibility = JavaVersion.VERSION_1_8

publishing {
    publications {
        maven(MavenPublication) {
            from(components.java)
        }
    }
}

tasks.withType(JavaCompile) {
    options.encoding = 'UTF-8'
}

```

**Comment by dsyer** [comment]
Fixed in f015573 (we don't need wro4j any more).
